bin/sign.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -e
+
+# This script uses AMO (addons.mozilla.org) API
+# to upload devtools add-on and get it signed.
+
+XPI=$1
+if [[ -z $XPI || ! -f $XPI ]]; then
+  echo "$0 expects path to xpi as first argument"
+  exit
+fi
+if [[ ! -x $(command -v jq) ]]; then
+  echo "$0 requires 'jq' to be installed"
+  exit
+fi
+if [[ ! -x $(command -v openssl) ]]; then
+  echo "$0 requires 'openssl' to be installed"
+  exit
+fi
+if [ -z $AMO_USER ]; then
+  echo "$0 expects AMO_USER env variable to be set"
+  exit
+fi
+if [ -z $AMO_SECRET ]; then
+  echo "$0 expects AMO_SECRET env variable to be set"
+  exit
+fi
+SCRIPT_DIR=$(dirname $0)
+
+VERSION=$(unzip -c $XPI install.rdf | grep em:version | grep -oE "[0-9.ab-]+")
+
+echo "Addon version: $VERSION"
+
+echo "Uploading xpi to sign"
+JWT=$($SCRIPT_DIR/jwt.sh)
+echo "JWT=$JWT"
```

Comment:
It is bad but not as bad as leaking the private key as the JSON web tokens are unique and contains seeds on each new token.